google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentConfigurationPanel.java

```diff
@@ -148,14 +148,22 @@ public void setDeploymentProjectAndVersion(DeploymentSource deploymentSource) {
     }
   }
 
+  /**
+   * Updates the text of the panel as follows:
+   *   if the project text box is empty, no message is displayed,
+   *   if the project text represents a valid project, the project details are displayed,
+   *   if the project text represents an invalid project, an error message is displayed.
+   */
   private void refreshApplicationInfoPanel() {
-    if (projectSelector.getProject() != null && projectSelector.getSelectedUser() != null) {
+    if (projectSelector.getText().isEmpty()) {
+      applicationInfoPanel.setMessage("", false /* isError*/);
```

Comment:
why do you need to set an empty message to indicate no message?

Reply:
I am not actually sure if this method is only called the first time the panel is loaded. It is also called from a public method, so I covered all cases.